contributors/design-proposals/minreadyseconds.md

```diff
@@ -0,0 +1,73 @@
+# Introduce Available Pods
+
+Today, Deployments, ReplicaSets, and DaemonSets use MinReadySeconds in order to include
+an additional delay on top of readiness checks and facilitate more robust rollouts. The
+ReplicaSet controller decides how many available Pods a ReplicaSet runs and the Deployment
+controller, when rolling out new Pods, will not proceed if the minimum available Pods that
+are required to run are not ready for at least MinReadySeconds (if MinReadySeconds is not
+specified then the Pods are considered available as soon as they are ready).
+
+Two problems that have been identified so far with the current state of things:
+1. A Pod is marked ready by the kubelet as soon as it passes its readiness check. The
+ReplicaSet controller runs as part of master and estimates when a Pod is available by
+comparing the time the Pod became ready (as seen by the kubelet) with MinReadySeconds.
+[Clock-skew between master and nodes will affect the availability checks](https://github.com/kubernetes/kubernetes/issues/29229). 
+2. PodDisruptionBudget is working with ready Pods and has no notion of MinReadySeconds
+when used by a Deployment/ReplicaSet. 
+
+Both problems above can be solved by moving MinReadySeconds in the PodSpec. Once kubelet
+observes that a Pod has been ready for at least MinReadySeconds without any of its containers
+crashing, it will update the PodStatus with an Available condition set to Status=True.
+Higher-level orchestrators running on different machines such as the ReplicaSet or the
+PodDisruptionBudget controller will merely need to look at the Available condition that
+is set in the status of a Pod.
+
+## API changes
+
+A new field is proposed in the PodSpec:
+```go
+	// Minimum number of seconds for which a newly created pod should be ready
+	// without any of its container crashing, for it to be considered available.
+	// Defaults to 0 (pod will be considered available as soon as it is ready)
+	// +optional
+	MinReadySeconds *int32 `json:"minReadySeconds,omitempty"`
```

Comment:
You must be "Ready" for "X Seconds" in order to be "Stable".  "StableAfterReadySeconds" is awkward but is somewhat closer.